src/client.rs

```diff
@@ -17,6 +17,41 @@ use compression::Compression;
 use authenticators::Authenticator;
 use error;
 
+macro_rules! builder_field {
+    ($field:ident, $field_type:ty) => {
+        pub fn $field<'a>(&'a mut self,
+                          $field: $field_type) -> &'a mut Self {
+            self.$field = Some($field);
+            self
+        }
+    };
+}
+
+/// Query Builder
+#[derive(Debug,Default)]
+pub struct QueryBuilder {
+    query: Option<String>,
+    consistency: Option<Consistency>,
+    values: Option<Vec<Value>>,
+    with_names: Option<bool>,
+    page_size: Option<i32>,
+    paging_state: Option<CBytes>,
+    serial_consistency: Option<Consistency>,
+    timestamp: Option<i64>
+}
+
+impl QueryBuilder {
```

Comment:
It is a. Builder pattern sort of ; e.g: https://doc.rust-lang.org/book/method-syntax.html